src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/Storage/DiskBasedResultStore.cs

```diff
@@ -0,0 +1,362 @@
+
+            await JsonSerializer.SerializeAsync(
+                stream,
+                result,
+                SerializerContext.Default.ScenarioRunResult,
+                cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc/>
+    public ValueTask DeleteResultsAsync(
+        string? executionName = null,
+        string? scenarioName = null,
+        string? iterationName = null,
+        CancellationToken cancellationToken = default)
+    {
+        if (executionName is null && scenarioName is null && iterationName is null)
+        {
+            Directory.Delete(_resultsRootPath, recursive: true);
+            _ = Directory.CreateDirectory(_resultsRootPath);
+        }
+        else if (executionName is not null && scenarioName is null && iterationName is null)
+        {
+            var executionDir = new DirectoryInfo(Path.Combine(_resultsRootPath, executionName));
+
+            if (executionDir.Exists)
+            {
+                executionDir.Delete(recursive: true);
+            }
+        }
+        else if (executionName is not null && scenarioName is not null && iterationName is null)
+        {
+            var scenarioDir =
+                new DirectoryInfo(Path.Combine(_resultsRootPath, executionName, scenarioName));
+
+            if (scenarioDir.Exists)
+            {
+                scenarioDir.Delete(recursive: true);
+            }
+        }
+        else if (executionName is not null && scenarioName is not null && iterationName is not null)
+        {
+            var resultFile =
+                new FileInfo(Path.Combine(_resultsRootPath, executionName, scenarioName, $"{iterationName}.json"));
+
+            if (resultFile.Exists)
+            {
+                resultFile.Delete();
+            }
+        }
+        else
+        {
+            IEnumerable<FileInfo> resultFiles =
+                EnumerateResultFiles(executionName, scenarioName, iterationName, cancellationToken);
+
+            foreach (FileInfo resultFile in resultFiles)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                DirectoryInfo scenarioDir = resultFile.Directory!;
+                DirectoryInfo executionDir = scenarioDir.Parent!;
+
+                resultFile.Delete();
+
+                if (!scenarioDir.EnumerateFileSystemInfos().Any())
+                {
+                    scenarioDir.Delete(recursive: true);
+
+                    if (!executionDir.EnumerateFileSystemInfos().Any())
+                    {
+                        executionDir.Delete(recursive: true);
+                    }
+                }
+            }
+        }
+
+        return default;
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<string> GetLatestExecutionNamesAsync(
+        int? count = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        await Task.CompletedTask.ConfigureAwait(false);
```

Comment:
Better to just suppress the warning rather than add unnecessary awaits

Reply:
Fixed